code_samples/front/shop/list_products_in_catalog/src/Controller/PromoController.php

```diff
@@ -0,0 +1,33 @@
+<?php
+
+namespace App\Controller;
+
+use Ibexa\Bundle\Core\Controller;
+use Ibexa\Contracts\ProductCatalog\CatalogServiceInterface;
+use Ibexa\Contracts\ProductCatalog\Values\Product\ProductQuery;
+use Ibexa\Contracts\ProductCatalog\ProductServiceInterface;
+
+final class PromoController extends Controller
+{
+    private ProductServiceInterface $productService;
+
+    private CatalogServiceInterface $catalogService;
+
+    public function __construct(ProductServiceInterface $productService, CatalogServiceInterface $catalogService)
+    {
+        $this->productService = $productService;
+        $this->catalogService = $catalogService;
+    }
+
+    public function listPromoProductsAction()
+    {
+        $catalog = $this->catalogService->getCatalogByIdentifier('desk_promo');
+        $query = new ProductQuery($catalog->getQuery());
+
+        $result = $this->productService->findProducts($query);
```

Comment:
Maybe we can add pagination/sorting to the example?

Reply:
Given that this is for illustrative purposes, I prefer to keep the examples as simple as possible, even if they are not actual best practices